packages/elements/src/components/file-picker/file-picker.component.ts

```diff
@@ -0,0 +1,288 @@
+
+    if (changedProperties.has('rootModel')) {
+      this.#prepareMap();
+      this.#filter('');
+    }
+  }
+
+  open() {
+    this.openPicker = true;
+
+    void this.updateComplete.then(() => this.#focusInput());
+  }
+
+  render() {
+    if (!this.openPicker) {
+      return nothing;
+    }
+
+    return html`
+      <div
+        id="backdrop"
+        @click="${() => this.#closePicker()}"
+        class="fixed left-0 top-0 z-50 flex h-full w-full justify-center bg-black/50 backdrop-blur-lg"
+      >
+        <div
+          @click="${(e: MouseEvent) => e.stopPropagation()}"
+          role="dialog"
+          id="picker"
+          class="m-4 flex h-fit max-h-[500px] w-full max-w-[40rem] flex-col rounded-lg bg-gray-200/60 p-4 backdrop-blur-lg md:w-1/2"
+        >
+          <div class="mb-2 flex items-center rounded bg-gray-200/60 p-2 backdrop-blur-lg">
+            <div class="mx-2 flex items-center text-gray-800">${searchIcon}</div>
+            <label for="file-picker-input" class="sr-only">Search for a file</label>
+            <input
+              id="file-picker-input"
+              @keyup="${(e: KeyboardEvent) => this.#handleSearch(e)}"
+              type="search"
+              style="box-shadow: none"
+              class="mr-2 w-full rounded border-0 border-transparent bg-transparent text-gray-800 focus:shadow-none"
+              placeholder="Search for a file (Ctrl-K)"
+            />
+          </div>
+          <div tabindex="-1" class="overflow-auto"> ${this.#renderFoundFiles()} </div>
+        </div>
+      </div>
+    `;
+  }
+
+  #renderFoundFiles() {
+    return html`
+      <ul id="files" class="flex flex-col">
+        ${renderIf(this.filteredFiles.length === 0, () => html`<li class="text-gray-800">No files found</li>`)}
+        ${repeat(
+          this.filteredFiles,
+          (item) => item.name,
+          ({ name, file }, index) => {
+            const view = this.#getView(file);
+            return html`
+              <li>
+                <a
+                  ?data-active="${index === this.fileIndex}"
+                  tabindex="${index === this.fileIndex ? 0 : -1}"
+                  @focusout="${() => this.#handleFocus()}"
+                  @click="${() => this.#closePicker()}"
+                  class="my-1 flex flex-wrap items-center rounded border-2 border-white bg-white p-1 px-2 text-gray-800 outline-none data-[active]:border-primary-500"
+                  href="${toAbsoluteUrl(view, name)}"
+                >
+                  <span class="inline-flex" title="File with ${view}s">${this.#renderTestOrMutantIndication(view)}</span>
+                  <span class="ms-1">${file.result?.name}</span>
+                  <span class="mx-2">•</span>
+                  <span class="text-gray-400">${name}</span>
+                </a>
+              </li>
+            `;
+          },
+        )}
+      </ul>
+    `;
+  }
+
+  #renderTestOrMutantIndication(view: View) {
+    return html`${when(
```

Comment:
Why the `when` call inside a html template tag? Isn't a ternary just simpler? :D